### DISCUSSION TITLE
Customize user ID field name?

### DISCUSSION THREAD

**Discussion by JustinDigsData** [fastapi-users/fastapi-users#1376] [Q&A] [answered]
Hi,

I'm trying to port code and an existing database over to using FastAPI Users for authentication and am running into one small issue around the name of the ID column.

My existing user database's ID column is called "user_id" instead of just "id." Is there a way to override the default ID column name? Right now, I'm getting an AttributeError when I attempt to use that column in my User class:

`  File ".venv\Lib\site-packages\fastapi_users_db_sqlalchemy\__init__.py", line 120, in get
    statement = select(self.user_table).where(self.user_table.id == id)
                                              ^^^^^^^^^^^^^^^^^^
AttributeError: type object 'User' has no attribute 'id'`

**Comment by JustinDigsData** [comment]
Just saw this note: https://github.com/fastapi-users/fastapi-users/discussions/1245

Assuming it's still the case that the name of the User id field just is what it is?

**Comment by frankie567** [comment, marked as answer]
Well, if it's just the name of the column in the database; but it's okay for you to use `id` in the Python code, you can either:

* Make a migration on your database to rename the column
* Explicitly tell SQLAlchemy the name of the column:

```py
import uuid

from fastapi_users.db import SQLAlchemyBaseUserTable
from fastapi_users_db_sqlalchemy.generics import GUID
from sqlalchemy.orm import Mapped, mapped_column


class User(SQLAlchemyBaseUserTable[uuid.UUID], Base):
    id: Mapped[uuid.UUID] = mapped_column(
        "user_id", GUID, primary_key=True, default=uuid.uuid4
    )
```

*(assuming your `id` is an UUID, but it can be another thing, like a integer)*

**Reply by JustinDigsData** [reply]
Perfect, thanks. I have quite a few other references throughout, so probably easiest just for me to do a db migration... but good to know that the Mapped[] option is there for future reference.
